exec/stages/stages.go

```diff
@@ -31,29 +30,20 @@ type StageCreator interface {
 	Name() string
 }
 
-var stages map[string]StageCreator
+var stages = registry.Create("stages")
 
 func Register(stage StageCreator) {
-	if stages == nil {
-		stages = map[string]StageCreator{}
-	}
-	if _, ok := stages[stage.Name()]; ok {
-		panic(fmt.Sprintf("stage %q already registered", stage.Name()))
-	}
-	stages[stage.Name()] = stage
+	stages.Register(stage)
 }
 
 func Get(name string) StageCreator {
-	if stage, ok := stages[name]; ok {
-		return stage
+	s := stages.Get(name)
+	if s == nil {
+		return nil
 	}
-
-	return nil
+	return s.(StageCreator)
```

Comment:
Yeah, this should be fine. When I looked at this last (on my phone), I couldn't see the full context.